api_tests/registrations/views/test_registration_detail.py

```diff
@@ -1,297 +1,293 @@
+        assert res.status_code == 403
+
+    #   test_update_private_registration_logged_in_read_write_contributor
+        res = app.put_json_api(private_url, private_registration_payload, auth=read_write_contributor.auth, expect_errors=True)
+        assert res.status_code == 403
+
+    #   test_update_public_registration_to_private
+        public_to_private_payload = make_payload(id=public_registration._id, attributes={'public': False})
+
+        res = app.put_json_api(public_url, public_to_private_payload, auth=user.auth, expect_errors=True)
+        assert res.status_code == 400
+        assert res.json['errors'][0]['detail'] == 'Registrations can only be turned from private to public.'
+
+    def test_fields(self, app, user, public_registration, private_registration, public_url, private_url, make_payload):
+
+    #   test_public_field_has_invalid_value
+        invalid_public_payload = make_payload(id=public_registration._id, attributes={'public': 'Dr.Strange'})
+
+        res = app.put_json_api(public_url, invalid_public_payload, auth=user.auth, expect_errors=True)
+        assert res.status_code == 400
+        assert res.json['errors'][0]['detail'] == '"Dr.Strange" is not a valid boolean.'
+
+    #   test_fields_other_than_public_are_ignored
+        attribute_list = {
+            'public': True,
+            'category': 'instrumentation',
+            'title': 'New title',
+            'description': 'New description'
         }
+        verbose_private_payload = make_payload(attributes=attribute_list)
 
-        url = '/{}registrations/{}/'.format(API_BASE, node1._id)
-        res = self.app.put_json_api(url, payload, auth=self.user.auth)
-        assert_equal(res.json['data']['attributes']['public'], True)
-        node1.reload()
-        assert_true(node1.is_public)
+        res = app.put_json_api(private_url, verbose_private_payload, auth=user.auth)
+        assert res.status_code == 200
+        assert res.json['data']['attributes']['public'] == True
+        assert res.json['data']['attributes']['category'] == 'project'
+        assert res.json['data']['attributes']['description'] == private_registration.description
+        assert res.json['data']['attributes']['title'] == private_registration.title
+
+    #   test_type_field_must_match
+        node_type_payload = make_payload(type='node')
+
+        res = app.put_json_api(private_url, node_type_payload, auth=user.auth, expect_errors=True)
+        assert res.status_code == 409
+
+    #   test_id_field_must_match
+        mismatch_id_payload = make_payload(id='12345')
+
+        res = app.put_json_api(private_url, mismatch_id_payload, auth=user.auth, expect_errors=True)
+        assert res.status_code == 409
+
+    def test_turning_private_registrations_public(self, app, user, make_payload):
+        private_project = ProjectFactory(creator=user, is_public=False)
+        private_registration = RegistrationFactory(project=private_project, creator=user, is_public=False)
```

Comment:
Keep it for now--there were some old registrations that were allowed to be made private, so we do want to support this functionality.